### PR TITLE
Screen generator

### PR DESCRIPTION
**Pull request by pabloferro**
## Summary
Add screen generator

## Screenshots
https://user-images.githubusercontent.com/4098152/110337756-0874db80-8005-11eb-9e81-9a5caea6bd43.mp4

## Possible Improvements
* Add prompt for context
* Add prompt for authenticated route

### CODE REVIEW

File: generators/app/templates/plopfile.js [hunk 1]
```diff
@@ -11,19 +11,98 @@ module.exports = function plopConfig(plop) {
...
+        type: 'add',
+        path: 'src/screens/{{name}}/styles.module.scss',
+        templateFile: 'templates/screen/styles.module.scss.hbs'
+      },
+      {
+        type: 'modify',
+        path: 'src/components/Routes/constants.ts',
+        pattern: '// Add imports for screens above (FOR GENERATORS, DO NOT REMOVE)',
+        template:
+          "const {{name}} = lazy(() => import('../../screens/{{name}}'));\n// Add imports for screens above (FOR GENERATORS, DO NOT REMOVE)"
```
Comment (damfinkel) [inline generators/app/templates/plopfile.js:84]: what's this comment?

File: generators/app/templates/plopfile.js [hunk 2]
```diff
@@ -11,19 +11,98 @@ module.exports = function plopConfig(plop) {
...
+        type: 'append',
+        path: 'src/components/Routes/constants.ts',
+        pattern: 'export const ROUTES = [',
+        template: `  {
+    exact: false,
+    path: PATHS.{{camelCase name}},
+    component: {{name}},
+    title: 'Routes:{{camelCase name}}Title',
+    description: 'Routes:{{camelCase name}}Description'
+    // redirectTo: (user: Nullable<User>) => (user ? undefined : MAIN_PUBLIC_PATH)
```
Comment (damfinkel) [inline generators/app/templates/plopfile.js]: Why is this commented? Can we ask the user if they want a public or private route?

Comment (pabloferro) [reply generators/app/templates/plopfile.js]: yes

File: generators/app/templates/plopfile.js (hunk 1 above)
Comment (pabloferro) [reply generators/app/templates/plopfile.js:84]: It's just to add the new imports at the end 😅 . In the generator I search for that comment and add the route above it.

**Review by alejobh — APPROVED** [review]

**Review by damfinkel — APPROVED** [review]

**Pull request merged**